packages/api/test/functional/user-fields-checking.spec.js

```diff
@@ -0,0 +1,352 @@
+		),
+	);
+});
+
+test('POST /ideas checks user personal data before creates a new idea', async ({ client }) => {
+	const { user: loggedUser } = await createUser({
+		append: { address: null },
+	});
+	await loggedUser.institution().dissociate();
+	await loggedUser.areas().detach();
+
+	const unCompletedFields = ['address'];
+	const keywordTaxonomy = await Taxonomy.getTaxonomy('KEYWORDS');
+	const keywordTerms = await Factory.model('App/Models/Term').createMany(5);
+	await keywordTaxonomy.terms().saveMany(keywordTerms);
+	const keywordTermsIds = keywordTerms.map((keyword) => keyword.id);
+
+	const response = await client
+		.post('/ideas')
+		.loginVia(loggedUser, 'jwt')
+		.send({
+			title: 'wonderfull idea',
+			description: 'wonderfull idea description',
+			keywords: keywordTermsIds,
+		})
+		.end();
+
+	response.assertStatus(403);
+	response.assertJSONSubset(
+		errorPayload(
+			errors.REGISTRATION_UNCOMPLETED,
+			antl('error.user.registrationUncompleted', { unCompletedFields }),
+		),
+	);
+});
+
+test('POST /services checks user personal and organizational data before creates a new Service', async ({
+	client,
+}) => {
+	const { user: loggedUser } = await createUser({
+		append: { address: null },
+	});
+	await loggedUser.institution().dissociate();
+	await loggedUser.areas().detach();
+
+	const unCompletedFields = ['address', 'institution'];
+
+	const serviceFactory = await Factory.model('App/Models/Service').make();
+
+	const keywordTaxonomy = await Taxonomy.getTaxonomy('KEYWORDS');
+	const keywordTerms = await Factory.model('App/Models/Term').createMany(5);
+	await keywordTaxonomy.terms().saveMany(keywordTerms);
+	const keywordTermsIds = keywordTerms.map((keyword) => keyword.id);
+
+	const response = await client
+		.post('/services')
+		.loginVia(loggedUser, 'jwt')
+		.send({
+			...serviceFactory.toJSON(),
+			keywords: keywordTermsIds,
+		})
+		.end();
+
+	response.assertStatus(403);
+	response.assertJSONSubset(
+		errorPayload(
+			errors.REGISTRATION_UNCOMPLETED,
+			antl('error.user.registrationUncompleted', { unCompletedFields }),
+		),
+	);
+});
+
+test('POST /services/orders checks user personal data before creates a new Service Order', async ({
+	client,
+}) => {
+	const { user: loggedUser } = await createUser({
+		append: { address: null },
+	});
+	await loggedUser.institution().dissociate();
+	await loggedUser.areas().detach();
+
+	const unCompletedFields = ['address'];
+
+	const { user: responsible } = await createUser({ append: { status: 'verified' } });
+
+	const servicesFactory = await Factory.model('App/Models/Service').makeMany(3);
+	const services = await responsible
+		.services()
+		.createMany(servicesFactory.map((service) => service.toJSON()));
+
+	const serviceList = services.map((service) => {
+		return {
+			service_id: service.id,
+			quantity: 2,
+		};
+	});
+
+	const response = await client
+		.post('/services/orders')
+		.loginVia(loggedUser, 'jwt')
+		.send({
+			services: serviceList,
+		})
+		.end();
```

Comment:
Previnindo um erro futuro com minha PR 🙂
```suggestion
	const services = await Factory.model('App/Models/Service').createMany(3);
	await Promise.all([services.map((service) => service.user().associate(responsible))]);
	const payload = services.map((service) => ({ service_id: service.id, quantity: 2 }));

	const response = await client
		.post('/services/orders')
		.loginVia(loggedUser, 'jwt')
		.send({ services: payload })
		.end();
```